charts/mimir-distributed/INTERNAL.md

```diff
@@ -1,8 +1,36 @@
-# Notes on testing mimir-distributed
+# Developer notes
+
+## Notes on the kube manifest templates and configuration
+
+Why do we generate Kubernetes resource definitions from named templates, and not directly ? That is, why do we have
+```yaml
+{{- if not .Values.skipResourceRendering -}}
+{{- include "mimir.querierDeployment" . -}}
+{{- end -}}
+```
+instead of just writing the Deployment template directly ?
+
+[PR #1182](https://github.com/grafana/helm-charts/pull/1182) commit message, quote:
+
+When Mimir is used as a subchart, the rendering of the annotation "checksum/config" on PODs is wrong. This is important to trigger service restart when the config is changed.
+
+There are two options: config is in Mimir or shared with Mimir in which case it can contain templates and values of the parent chart, making it impossible to evaluate in the subchart. Or the config lives in the parent chart, but then it is not possible to reach it in Mimir as subchart and the checksum is checksum of empty string. See: "few important details" in [Helm subcharts and globals](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/)
```

Comment:
```suggestion
There are two options:
- config is in `mimir-distributed` or shared with `mimir-distributed` in which case it can contain templates and values of the parent chart, making it impossible to evaluate in the subchart.
- config lives in the parent chart, but then it is not possible to reach it in `mimir-distributed` as subchart and the checksum is checksum of empty string. See: "few important details" in [Helm subcharts and globals](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/)
```